lib/trino-filesystem-azure/src/main/java/io/trino/filesystem/azure/AzureFileSystem.java

```diff
@@ -445,8 +447,8 @@ private boolean isHierarchicalNamespaceEnabled(AzureLocation location)
             throws IOException
     {
         try {
-            DataLakeFileSystemClient fileSystemClient = createFileSystemClient(location);
-            return fileSystemClient.getDirectoryClient("/").exists();
+            BlobContainerClient client = createBlobContainerClient(location);
+            return client.getAccountInfo(Duration.of(3, SECONDS)).isHierarchicalNamespaceEnabled();
```

Comment:
how did you come up with the timeout value?

Reply:
It doesn't work so :)